Library/Homebrew/keg_relocate.rb

```diff
@@ -157,9 +157,13 @@ def find_dylib(bad_name)
   end
 
   def mach_o_files
+    hardlinks = Set.new
     mach_o_files = []
     path.find do |pn|
       next if pn.symlink? || pn.directory?
+      # if we've already processed a file, ignore its hardlinks (which have the same dev ID and inode)
+      # this prevents relocations from being performed on a binary more than once
+      next unless hardlinks.add? [pn.stat.dev, pn.stat.ino]
```

Comment:
I'd move this even further down, such that `hardlinks` only contains Mach-O files. No need to keep track of hard links of *all* the files in the keg.